streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -113,7 +113,9 @@ public StreamsProducer(final StreamsConfig config,
                 final String applicationId = config.getString(StreamsConfig.APPLICATION_ID_CONFIG);
                 producerConfigs.put(
                     ProducerConfig.TRANSACTIONAL_ID_CONFIG,
-                    applicationId + "-" + Objects.requireNonNull(processId, "processId cannot be null for exactly-once beta"));
+                    applicationId + "-" +
+                        Objects.requireNonNull(processId, "processId cannot be null for exactly-once beta") +
+                        "-" + threadId.split("-StreamThread-")[1]);
```

Comment:
Include the thread-index in the `transactional.id` to generate unique IDs for different threads within the same process.

(I stripped out the `"-StreamThread-"` string in contrast to the old code for shorten the ID)